templates/graph-theory.md

```diff
@@ -12,10 +12,107 @@ keywords:
   - topological sort
 ---
 
+
+
+### Depth First Search
+
+Go to [Tutorial](../tutorials/graph-theory/depth-first-search)
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+
+```cpp
+// Your code goes here
+```
+
+</TabItem>
+<TabItem value="py" label="Python">
+
+```py
+def dfs(node):
+    if(node == None):
+        return
+        
+    dfs(node.left)
+    
+    print("I just visited the left branch!")
+    print("I am number: " + str(node.val))
+    print("I am visiting the right branch now!")
+    
+    dfs(node.right)
```

Comment:
what you're doing here is called inorder traversal. actually it's hard be generalised as dfs.

Reply:
I removed the empty `TabItem` .
The tutorial for preorder and postorder traversal isn't available.What should I do?
Should I open a issue for that?

Reply:
You can find it [here](https://wingkwong.github.io/leetcode-the-hard-way/tutorials/graph-theory/binary-tree#traversal).

You may help include them there and add your python version based one that.

Reply:
I did the mentioned changes.